src/lib/block_producer/block_producer.ml

```diff
@@ -165,7 +165,8 @@ let report_transaction_inclusion_failures ~logger failed_txns =
 let generate_next_state ~zkapp_cmd_limit ~constraint_constants
     ~previous_protocol_state ~time_controller ~staged_ledger ~transactions
     ~get_completed_work ~logger ~(block_data : Consensus.Data.Block_data.t)
-    ~winner_pk ~scheduled_time ~log_block_creation ~block_reward_threshold =
+    ~winner_pk ~scheduled_time ~log_block_creation ~block_reward_threshold
+    ~zkapps_per_block_hard_cap =
```

Comment:
I don't like that two closely related parameters (`zkapp_cmd_limit` and `zkapps_per_block_hard_cap` are so distinctly different in their names. Suggest to name the new parameter as `zkapps_cmd_limit_hardcap` or something all over the codebase